pkg/reconciler/revision/resources/deploy.go

```diff
@@ -36,49 +36,88 @@ import (
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/intstr"
+
+	apiconfig "knative.dev/serving/pkg/apis/config"
 )
 
-//nolint:gosec // Filepath, not hardcoded credentials
-const concurrencyStateTokenVolumeMountPath = "/var/run/secrets/tokens"
-const concurrencyStateTokenName = "state-token"
+// VolumeNames
+//nolint:gosec // VolumeName, not hardcoded credentials
+const (
+	tokenVolumeName   = "knative-token-volume"
+	podInfoVolumeName = "pod-info"
+	certVolumeName    = "server-certs"
+	logVolumeName     = "knative-var-log"
+
+	PodInfoFeatureKey     = "features.knative.dev/podspec-podinfo"
+	PodInfoFeatureEnabled = "enabled"
+)
 
 var (
+	// Volumes
 	varLogVolume = corev1.Volume{
-		Name: "knative-var-log",
+		Name: logVolumeName,
 		VolumeSource: corev1.VolumeSource{
 			EmptyDir: &corev1.EmptyDirVolumeSource{},
 		},
 	}
 
-	varLogVolumeMount = corev1.VolumeMount{
-		Name:        varLogVolume.Name,
-		MountPath:   "/var/log",
-		SubPathExpr: "$(K_INTERNAL_POD_NAMESPACE)_$(K_INTERNAL_POD_NAME)_",
-	}
-
 	varTokenVolume = corev1.Volume{
-		Name: "knative-token-volume",
+		Name: tokenVolumeName,
 		VolumeSource: corev1.VolumeSource{
 			Projected: &corev1.ProjectedVolumeSource{
 				Sources: []corev1.VolumeProjection{{
 					ServiceAccountToken: &corev1.ServiceAccountTokenProjection{
 						ExpirationSeconds: ptr.Int64(600),
-						Path:              concurrencyStateTokenName,
+						Path:              queue.ConcurrencyStateTokenFilename,
 						Audience:          "concurrency-state-hook"},
 				}},
 			},
 		},
 	}
 
-	certVolumeMount = corev1.VolumeMount{
-		MountPath: queue.CertDirectory,
-		Name:      "server-certs",
+	varPodInfoVolume = corev1.Volume{
+		Name: podInfoVolumeName,
+		VolumeSource: corev1.VolumeSource{
+			DownwardAPI: &corev1.DownwardAPIVolumeSource{
+				Items: []corev1.DownwardAPIVolumeFile{{
+					Path: queue.PodInfoAnnotationsFilename,
+					FieldRef: &corev1.ObjectFieldSelector{
+						FieldPath: "metadata.annotations",
+					},
+				}},
+			},
+		},
+	}
+
+	varCertVolume = corev1.Volume{
+		Name: certVolumeName,
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{},
+		},
+	}
+
+	// Volume Mounts
+	varLogVolumeMount = corev1.VolumeMount{
```

Comment:
nit: not sure why this was moved but it make the diff smaller if it wasn't - same with some of the others

Reply:
moved for order and clarity - sort out a mesh of volumes and volumeMounts  - none critical either way

Reply:
I'd leave it for now to make this diff smaller